packages/venia-ui/lib/components/CheckoutPage/PaymentInformation/paymentMethods.gql.js

```diff
@@ -0,0 +1,23 @@
+import gql from 'graphql-tag';
+
+// We disable linting for local fields because there is no way to add them to
+// the fetched schema.
+// https://github.com/apollographql/eslint-plugin-graphql/issues/99
+/* eslint-disable graphql/template-strings */
+export const GET_SELECTED_PAYMENT_METHOD = gql`
+    query getSelectedPaymentMethod($cartId: String!) {
+        cart(cart_id: $cartId) {
+            id
+            selectedPaymentMethod @client
```

Comment:
The thing with getting `selected_payment_method` from `cart` is, we have to use the `setPaymentMethodOnCart` mutation but that mutation does not accept the payment nonce's details section. It only takes the nonce, but we need it to save the details as well like `type of the card and its last 4 digits`. If I make the change Ill have to maintain 2 diff versions, half will be in remote saving `selected_payment_method, nonce` and `payment_details` in the local.

This will create more confusion than needed. I would say we update the cart with the payment information it needs but also update it in the cache.

![image](https://user-images.githubusercontent.com/35203638/78920446-0606ac00-7a59-11ea-933a-df65f961f492.png)

The same goes for billing address. Remote does not store if the billing address is the same as the shipping address. It just returns you the address. Instead, we checking the addresses manually every time which is a time-consuming process, I would like to leave the boolean in the cache and update the appropriate address on the cart when the user submits.

Reply:
> The thing with getting selected_payment_method from cart is, we have to use the setPaymentMethodOnCart mutation but that mutation does not accept the payment nonce's details section. It only takes the nonce, but we need it to save the details as well like type of the card and its last 4 digits

I'd like to understand this a bit more - is the problem that we need to display type of card and last 4 digits to the user but we don't get that information from graphql?

I think it is reasonable to maintain and read from two different slices of state if you don't get everything back from the server query. I don't think you should be overloading the slice names though. `selectedPaymentMethod` is confusing since there is already a returned `selected_payment_method` in the response. 

> Remote does not store if the billing address is the same as the shipping address. It just returns you the address. Instead, we checking the addresses manually every time which is a time-consuming process

I see what you are saying here. That is confusing :/ I wish they set `same_as_shipping` in the response object of `cart.billing_address`.

Reply:
Yeah, the billing address data is being maintained at 2 diff sources. Billing address at remote and the boolean if billing is the same as shipping address in the local.

Also for the payment details. Payment nonce and selected payment method are stored at the remote and credit card details in the local.